script.js

```diff
@@ -1,4 +1,126 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+/* 
+User input = Scissors/Paper/Stone
+Output = Win/Lose
+    1        2       3        1
+Scissors > Paper > Stone > Scissors
+
+0-0 = 0
+0-1 = -1
+0-2 = -2
+
+1-0 = 1
+1-1 = 0
+1-2 = -1
+
+2-0 = 2
+2-1= 1
+2-2 = 0
+
+ */
+// Gobal Variables
+const selection = ["scissors", "paper", "stone"];
+var userWinLoseRecord = 0;
+var aiWinLoseRecord = 0;
+var totalRuns = 0;
+var userName = "";
+var userNameFlag = 0;
+
+var getRandomArray = function (lenOfArray) {
+  // Generate a decimal from 0 through 6, inclusive of 0 and exclusive of
+  var randomDecimal = Math.random() * lenOfArray;
+
+  // Remove the decimal with the floor operation.
+  // This will be an integer from 0 to 5 inclusive.
+  var randomInteger = Math.floor(randomDecimal);
+
+  return randomInteger;
+};
+
+var getAiOutput = function () {
```

Comment:
Appreciate how you put everything into functions outside of the main function. This is very good foundation for building beyond these smaller projects. Being able to break the program down into individual modules and put it together is really great!